src/react-components/room/ObjectsSidebarContainer.js

```diff
@@ -0,0 +1,42 @@
+import React, { useCallback, useRef } from "react";
+import PropTypes from "prop-types";
+import { ObjectsSidebar, ObjectsSidebarItem } from "./ObjectsSidebar";
+import { List } from "../layout/List";
+import { useObjectList } from "./useObjectList";
+
+export function ObjectsSidebarContainer({ onClose }) {
+  const listRef = useRef();
+  const { objects, selectedObject, selectObject, unfocusObject, focusObject } = useObjectList();
+
+  const onUnfocusListItem = useCallback(
+    e => {
+      if (e.relatedTarget === listRef.current || !listRef.current.contains(e.relatedTarget)) {
+        unfocusObject();
+      }
+    },
+    [unfocusObject, listRef]
+  );
+
+  return (
+    <ObjectsSidebar objectCount={objects.length} onClose={onClose}>
+      <List ref={listRef}>
+        {objects.map(object => (
+          <ObjectsSidebarItem
+            selected={selectedObject === object}
+            object={object}
+            key={object.id}
+            onClick={() => selectObject(object)}
+            onMouseOver={() => focusObject(object)}
+            onFocus={() => focusObject(object)}
+            onMouseOut={onUnfocusListItem}
+            onBlur={onUnfocusListItem}
+          />
+        ))}
+      </List>
+    </ObjectsSidebar>
```

Comment:
Design nit: It would be nice if the design made it clear that the objects list is empty. I'm thinking something like a large centered message, maybe with iconography, that says "There are no objects in the room". Right now you're just presented with a blank sidebar except for the title and the close button and close button at the top. It's especially important on mobile, where the blank space of the sidebar takes up the entire screen.